lisa/analysis/tasks.py

```diff
@@ -256,11 +256,53 @@ def stringify_row_state(row):
 
         task_state_df.rename(columns={'__cpu' : 'cpu'}, inplace=True)
         task_state_df = task_state_df[['target_cpu', 'cpu', 'curr_state']]
-        task_state_df['next_state'] = task_state_df.curr_state.shift(-1)
+        # Make the last "next_state" the same as the last "curr_state"
+        task_state_df['next_state'] = task_state_df.curr_state.shift(
+            -1, fill_value=task_state_df['curr_state'].values[-1])
         self.trace.add_events_deltas(task_state_df, inplace=True)
 
         return task_state_df
 
+    @classmethod
+    def stringify_df_task_states(cls, df_task_states, inplace=False):
+        """
+        Convert a Dataframe obtained from :meth:`df_task_states` into its
+        :meth:`df_task_states_stringified` variant.
+        """
+        df = df_task_states if inplace else df_task_states.copy()
+
+        def stringify_state(state):
+            try:
+                return TaskState(state).char
+            except ValueError:
+                return TaskState.sched_switch_str(state)
+
+        for col in ['curr_state', 'next_state']:
+            df[col] = df[col].apply(stringify_state)
+
+        return df
+
+    @df_task_states.used_events
+    def df_task_states_stringified(self, task):
```

Comment:
Why not keeping just `df_task_states()` and add there a `stringify=false` param ?

Reply:
True, I did not even consider this... Dunno which is better

Reply:
`stringify_state=False` would avoid duplicating the docstring indeed

Reply:
I personally like better polymorphic APIs... then multiple APIs... :)